assets/src/scripts/components/hydrograph/index.js

```diff
@@ -481,33 +487,31 @@ const dateRangeControls = function(elem, siteno) {
     li.select(`#${DATE_RANGE[0].label}`).attr('checked', true);
 };
 
-
-const noDataAlert = function(elem, tsCollectionIds) {
+const dataLoadingAlert = function(elem, message) {
     elem.select('#no-data-message').remove();
-    if (tsCollectionIds && tsCollectionIds.length === 0) {
+    if (message) {
         elem.append('div')
             .attr('id', 'no-data-message')
             .attr('class', 'usa-alert usa-alert-info')
             .append('div')
-                .attr('class', 'usa-alert-body')
-                .append('p')
-                    .attr('class', 'usa-alert-text')
-                    .text('No current time series data available for this site');
+            .attr('class', 'usa-alert-body')
+            .append('p')
+            .attr('class', 'usa-alert-text')
+            .text(message);
     }
 };
 
-export const attachToNode = function (store, node, {siteno, parameter, compare, cursorOffset, interactive = true} = {}) {
+export const attachToNode = function (store, node, {siteno, parameter, compare, period, cursorOffset, showOnlyGraph = false} = {}) {
```

Comment:
In addition to changes required for the period, I reordered the code to put fetching before the rendering code. Also reordered the rendering code so it's clearer what is being rendered. 

I think this module could use some decomposition (for instance moving the date controls into their own module and perhaps moving the chart into its own module) so that the index module is basically using functions from submodules. I think it would make testing easier. I am going to create a tech-debt JIRA to address this.